### ISSUE TITLE
Updating Nextcloud DB seems somewhat broken

### ISSUE THREAD

**Issue by redm123** [FrankHausmann/IOSKeePass#43]
I have my DB file on a Nextcloud instance. This generally works, but pulling an updated file seems somewhat broken. Not sure if this should happen automatically, but even manually it seems a bit weird ... or broken.

I have the Nextcloud iOS app installed.

So I updated the DB file and it's available on the Nextcloud. 
* it is not automatically picked up by IOSKeePass.
* I click on the icon at the bottom (square with arrow ... not very obvious, btw.)
* I select Nextcloud -> database file (I see it's modified)
**  it downloads, I think, but that's it. It's not picked up by IOSKeePass
* I pull down the app page, spinning wheel. However, it's not picked up by IOSKeePass
* I can't even remove the database, to re-add it (just nothing happens when I try)

The only way I found to update the DB file, but it's a bit awkward: 
* in the Nextcloud app, Open With: IOSKeePass
* this does not work, but the existing database in IOSKeePass disappears
* and now I can open the updated one from the Nextcloud app via square icon with arrow...

**Comment by redm123** [comment]
Oh and an added twist to this: If I have managed to update the DB file, it's not picked up by AutoFill.
* I have to diable/enable autofill
* forcekill and restart the app
* open DB and confirm autofill...

**Comment by FrankHausmann** [comment]
The synchronization with webdav, icloud or onedrive is quite simple.

Before that, a note about the import only oneaml import the kdbx from the cloud, if this is present it will not be overwritten.

It is better to turn on the syncronisiation, I admit the name "cloud backup is confusing". So please set the cloud backup to on in settings, then enter the access data in the user area.

If the access is set up, a cloud appears in the upper right corner of the fileviewer, if the KDB(x) is opened now, it will be uploaded and updated after saving and going back to the fileviewer, in your case to Nextcloud.

The other way if the KDB(x) is newer than this in Keepass, simply in the FileViewer on the appropriate KDB(x) a long press touch execute and Sync select, now first a KDB(x) back file on IOS is created and the KDB(x) with the new version from the Nextcloud overwritten, new is always this atei which has the latest modifications date.

Translated with www.DeepL.com/Translator (free version)

**Comment by redm123** [comment]
> It is better to turn on the syncronisiation, I admit the name "cloud backup is confusing". So please set the cloud backup to on in settings, then enter the access data in the user area.
> 
> If the access is set up, a cloud appears in the upper right corner of the fileviewer, if the KDB(x) is opened now, it will be uploaded and updated after saving and going back to the fileviewer, in your case to Nextcloud.
> 
Not sure I understand that... You mean this would directly sync may KDBX with my Nextcloud via Webdav? Ignoring/not using the iOS NC app? And even two-way? (I was only thinking about NC -> IOSKeepass updates for now... as I previously had with Minikeepass)

> The other way if the KDB(x) is newer than this in Keepass, simply in the FileViewer on the appropriate KDB(x) a long press touch execute and Sync select, now first a KDB(x) back file on IOS is created and the KDB(x) with the new version from the Nextcloud overwritten, new is always this file which has the latest modifications date.

Hmm, the FileViewer I reach by tapping this little square with arrow icon at the bottom (then Speicherorte -> Nextcloud)? And by Sync you mean downloading the updated file (there is this little cloud icon on the KDBX document if there is an update)? Then I think I tried this. The updated file is downloaded, but IOSkeepass does not pick it up. The DB entry on the IOSkeepass main screen still shows the old date and time (and the old content).

**Comment by FrankHausmann** [comment]
Yes , the sync works as described in both directions, in the direction of NC it works automatically after changing an entry and leaving the group to the FileViewer, a circle progress is shown which indicates the progress. The KDBx is then on NC under your account in the directory IOSKeePass->Backup, the path to the IOS must be triggered manually via Long press ->Sync in the FileViewer. I had already implemented this about 15 months ago, and had also written a small guide for It professionals in the GitHub issue. **See Continual Sync with OneDrive**, but it also works with iCloud and NextCloud. All UniComers use it, everyone else just complains instead of trying it out themselves, not everyone understands that opensource free has its limits.

**Comment by FrankHausmann** [comment]
No, that's not what I mean. 
Please proceed as in the first comment and it will work.
![IMG_9048](https://user-images.githubusercontent.com/67282017/208617516-72004cfa-586f-4fd8-93f5-6ba57495a73e.jpg)
![IMG_9049](https://user-images.githubusercontent.com/67282017/208617746-7d800287-c13d-452f-a4dc-a096308d77f6.jpg)
![IMG_9050](https://user-images.githubusercontent.com/67282017/208617877-92479c24-adcd-4388-89c0-e9f94dfb26aa.jpg)
![IMG_9051](https://user-images.githubusercontent.com/67282017/208618022-bc5f7e25-3f
![IMG_9052](https://user-images.githubusercontent.com/67282017/208618158-5846182f-32cc-491e-913b-c04cde9df920.jpg)
95-4304-8ce4-76fc69def70e.jpg)

**Comment by redm123** [comment]
Ok, I got it now! We've been talking about different things. Thanks for your help!

And even though I solved it in a different, and granted even better way, I think I still have a point with my initial problems. Like  when adding the file again (e.g. from my Nextcloud app) it should update the file in IOSKeepass, or pulling down should pick it up, or not being able to delete (via Edit menu). Just if you happen to look for things to do.. ;)

---
So for the record, in case somebody else might find it useful, that's how to setup the 2-way sync with a Webdav storage / desktop app:
* the setup is: 
  * a cloud storage, like a Nextcloud instance
  * a desktop with a folder sync to the cloud storage. Put the initial KDB(X) file into that synced folder.
  * a mobile device with IOSKeepass
* in IOSKeepass settings enable Cloud Backup, and enter Webdav URL and credentials, for instance of a Nextcloud instance
* after that(!) add the KDB(X) file to IOSKeepass. This can be done via the square-with-arrow button at the bottom of the main screen, which allows to access other locations, like also the Nextcloud app. This should show the initial KDB(X) from the synced folder.
  * in case you already added the file before, long press on the file and select Delete Keepass DB (delete via Edit menu does not work)
* now open the KDB(X) file, change something, and go back out to the main screen
* a circle animation should appear performing the upload, and briefly showing a message that the remote folder was created
* this should have created a new folder in the WebDAV storage called "IOSKeepass" / "Backaups" with the KDB(X) file inside
* now this folder needs to be included in the desktop sync, to make it accessible with your desktop Keepass application 
* open the newly synced KDB(X) file in you desktop Keepass application. It will be the primary sync point now. The initial file is not needed anymore.
* to sync changes back from the desktop, open IOSKeepass and long press on the KDB(X) file. A popup menu will appear with the option Sync. This will download the changed file to IOSKeepass.
---

> everyone else just complains instead of trying it out themselves, not everyone understands that opensource free has its limits.

Well, I did try and I also know what open source is. I tried what seemed more or less obvious, but none of this worked. Not everything that is obvious to you as developer is also obvious to users. And that path you described is actually quite hidden.
* the whole process itself as sync mechanism is somewhat backward 
* I have to know there is this long press option.
* after setting up Webdav URL and credentials I had to delete and re-add the file, otherwise there was only an unclear error message
* even after that the Sync does not upload the file. I have to change something in the file to trigger the initial upload.
* I have to sync the backup folder back to my desktop and change my desktop keepass app to work on that backup path

See, even with knowing the process, there are some caveats.
I mean, as you say it's open source. Fair enough, I don't expect anything from you! But it could be that people don't find this solution or think something is broken..

> I admit the name "cloud backup is confusing". 

Actually I think the term "backup" describes pretty well what it does currently. It copies the file from the device to the remote storage when it's modified. I can't use the mechanism to feed the file into IOSKeepass in the first place. So it IS an independent backup path. It's just that it can be repurposed afterwards as a sync mechanism... I think just calling it e.g. "Cloud Sync" wouldn't really make it a lot more obvious by itself.


Btw. I found that I can delete KDBX files with the long press menu, too. But if I use (the obvious) Bearbeiten in the Fileviewer, I get those red circle icons in front of the files, which normally let me remove entries. But here nothing happens if I tap them. And when I tap the KDBX which is now in the Papierkorb the app crashes (somehow the sync backups don't let it crash, though).

**Comment by FrankHausmann** [comment]
thank you for the description of the process
